### PR TITLE
Operator tests fix

### PR DESCRIPTION
**Pull request by eliasalveszup**
- Added validation on request to verify whether the specified namespace exists and is active
- FIxing 500 error when trying to access invalid github or gitlab chart url (changed to 400)

### CODE REVIEW

File: install/helm-chart/templates/butler-cluster-role.yaml [hunk 1]
```diff
@@ -13,4 +13,10 @@ rules:
   - delete
   - get
   - patch
-{{- end }}
\ No newline at end of file
+- apiGroups:
```
Comment (eliasalveszup) [inline install/helm-chart/templates/butler-cluster-role.yaml:16]: This cluster role needs to be addes to allow butler to get information about any namespace at root level

**Comment by lucasbfernandes** [comment]
Fix DCO

File: butler/src/app/v2/api/deployments/pipes/namespace-validation.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    @InjectRepository(DeploymentEntity)
+    private readonly deploymentsRepository: Repository<DeploymentEntity>,
+    private readonly k8sClient: K8sClient
+  ) {}
+
+  public async transform(deploymentRequest: CreateDeploymentRequestDto): Promise<CreateDeploymentRequestDto> {
+    
+    const response = await this.k8sClient.getNamespace(deploymentRequest.namespace)
+
+    if (response.body.status?.phase !== "Active") {
```
Comment (lucasbfernandes) [inline butler/src/app/v2/api/deployments/pipes/namespace-validation.pipe.ts:38]: Use single quotes, otherwise lint will fail

**Review by lucasbfernandes — CHANGES_REQUESTED** [review]

**Comment by eliasalveszup** [comment]
Closed due to another PR with DCO fix

**Pull request closed without merging**